state/upgrades_test.go

```diff
@@ -1819,3 +1772,58 @@ func (s *upgradesSuite) TestAddModelEnvironVersion(c *gc.C) {
 		expectUpgradedData{models, expectedModels},
 	)
 }
+
+func (s *upgradesSuite) checkAddPruneSettings(c *gc.C, ageProp, sizeProp string, updateFunc func(st *State) error) {
+	settingsColl, settingsCloser := s.state.getRawCollection(settingsC)
+	defer settingsCloser()
+	_, err := settingsColl.RemoveAll(nil)
+	c.Assert(err, jc.ErrorIsNil)
+
+	m1 := s.makeModel(c, "m1", testing.Attrs{
+		ageProp:  "96h",
+		sizeProp: "4G",
+	})
+	defer m1.Close()
+
+	m2 := s.makeModel(c, "m2", testing.Attrs{})
+	defer m2.Close()
+
+	err = settingsColl.Insert(bson.M{
+		"_id": "someothersettingshouldnotbetouched",
+		// non-model setting: should not be touched
+		"settings": bson.M{"key": "value"},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	cfg1, err := m1.ModelConfig()
+	c.Assert(err, jc.ErrorIsNil)
+	expected1 := cfg1.AllAttrs()
+	expected1["resource-tags"] = ""
+
+	cfg2, err := m2.ModelConfig()
+	c.Assert(err, jc.ErrorIsNil)
+	expected2 := cfg2.AllAttrs()
+	expected2[ageProp] = "336h"
```

Comment:
Is the default for action and history records the same?
Maybe better to pass in the default values as function params